src/osdc/ObjectCacher.cc

```diff
@@ -1957,20 +1959,26 @@ void ObjectCacher::flusher_entry()
 		   << target_dirty << " target, "
 		   << max_dirty << " max)"
 		   << dendl;
-    loff_t actual = get_stat_dirty() + get_stat_dirty_waiting();
 
     ZTracer::Trace trace;
     if (cct->_conf->osdc_blkin_trace_all) {
       trace.init("flusher", &trace_endpoint);
       trace.event("start");
     }
 
+    loff_t actual = get_stat_dirty() + get_stat_dirty_waiting();
     if (actual > 0 && (uint64_t) actual > target_dirty) {
       // flush some dirty pages
       ldout(cct, 10) << "flusher " << get_stat_dirty() << " dirty + "
 		     << get_stat_dirty_waiting() << " dirty_waiting > target "
 		     << target_dirty << ", flushing some dirty bhs" << dendl;
       flush(&trace, actual - target_dirty);
+    } else if (dirty_or_tx_bh.size() > target_dirty_bh) {
```

Comment:
Yes, but it appears to be added to the number of BHs (`max_dirty_bh`) in `_maybe_wait_for_writeback()`.  I'm not quite sure what the logic is supposed to be here, but since the root cause of the issue appears to be the discrepancy between the if condition in `_maybe_wait_for_writeback()` and the if conditions in `flusher_entry()`, I wanted to highlight it.

Reply:
hi，I added it for consistency now